src/components/ActivityView.vue

```diff
@@ -90,19 +92,28 @@ function calculateCurrencyValue(valueInCrypto: bigint) {
 
 function calculateTotal(activity: Activity) {
   if (activity.transaction) {
-    return (
-      activity.transaction.amount +
-      activity.transaction.gasUsed * activity.transaction.gasPrice
-    )
+    const gasUsed = activity.transaction?.gasUsed || 0n
+    const gasPrice = activity.transaction?.gasPrice || 0n
+    return activity.transaction.amount + gasUsed * gasPrice
   }
   return 0n
 }
 
 function getAmount(amount: bigint, isGas = false) {
-  if (isGas) {
-    return beautifyBalance(Number(ethers.utils.formatUnits(amount, 'gwei')), 5)
+  if (app.chainType === ChainType.solana_cv25519) {
```

Comment:
Can we genericize this? Looks like we are hard-coding these conversions from the lowest denomination to the base denomination everywhere. Will be unscalable if we implement even more chain types.

Reply:
Fixed here [ba5fee8](https://github.com/arcana-network/wallet-ui/pull/535/commits/ba5fee89321cfaab746e450f6d3411ae17261c15)

Reply:
Fixed here https://github.com/arcana-network/wallet-ui/pull/535/commits/ba5fee89321cfaab746e450f6d3411ae17261c15